roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/util/ConfigParser.java

```diff
@@ -87,21 +88,15 @@ public IRoundaboutStructure generateStructure(RoundAboutConfig roundAboutConfig,
         return roundaboutStructure;
     }
 
-    private void addConnector(Set<StreetConnector> connectors, Set<Street> startPreviousSections, Set<Street> startNextSections, Set<Street> endPreviousSections, Set<Street> endNextSections, int iterationCount, int lastIteration) {
-        if (iterationCount == lastIteration) {
-            StreetConnector streetConnector = new StreetConnector(startPreviousSections, startNextSections);
-            connectors.add(streetConnector);
-            StreetConnector streetConnectorEnd = new StreetConnector(endPreviousSections, endNextSections);
-            connectors.add(streetConnectorEnd);
-        } else {
-            StreetConnector streetConnector = new StreetConnector(startPreviousSections, startNextSections);
-            connectors.add(streetConnector);
-            startNextSections.clear();
-            startNextSections.clear();
+    private void generateParameters(IRoundaboutStructure roundaboutStructure, RoundAboutConfig roundaboutConfig) {
+        Iterator<Parameter> keyIterator = roundaboutConfig.getRoundabout().getParameters().getParameter().iterator();
```

Comment:
a bit hard to understand, "getParameters().getParameter().iterator()" 

i dont know if this is possible, but "getParameters().iterator()" seems more understandable to me

Reply:
this is due to the xml structure:
<parameters>
   <parameter />
  ...